src/plugins/analysis/users_and_passwords/code/password_file_analyzer.py

```diff
@@ -59,27 +75,50 @@ def _generate_analysis_entry(self, passwd_entries, uid: str):
+            result_entry = result['{}:unix'.format(key)] = {}
+            result_entry['type'] = 'unix'
+            result_entry['entry'] = b':'.join(entry).decode(encoding='utf_8', errors='replace')
             try:
                 if entry[1][0] == ord('$'):
-                    result[key]['password-hash'] = entry[1].decode(encoding='utf_8', errors='replace')
-                    cracked_pw = self._crack_hash(entry, result, key)
-                    result[key]['cracked'] = bool(cracked_pw)
+                    result_entry['password-hash'] = entry[1].decode(encoding='utf_8', errors='replace')
+                    cracked_pw = self._crack_hash(entry, result_entry)
+                    result_entry['cracked'] = bool(cracked_pw)
             except (IndexError, AttributeError, TypeError):
                 logging.warning('Unsupported Format: {}'.format(uid), exc_info=True)
         return result
 
-    def _crack_hash(self, passwd_entry, result_dict, key):
+    def _generate_mosquitto_entry(self, passwd_entries):
+        result = {}
+        for entry in [m.split(b'$') for m in passwd_entries]:
+            user = entry[0].decode(encoding='utf_8', errors='replace')[:-1]
+            salt_hash = entry[2].decode(encoding='utf_8', errors='replace')
+            passwd_hash = entry[3].decode(encoding='utf_8', errors='replace')
+            passwd_entry = '{}:$dynamic_82${}$HEX${}'.format(user, b64decode(passwd_hash).hex(), b64decode(salt_hash).hex())
+            result_entry = result['{}:mosquitto'.format(user)] = {}
+            result_entry['type'] = 'mosquitto'
+            result_entry['entry'] = b'$'.join(entry).decode(encoding='utf_8', errors='replace')
+            result_entry['password-hash'] = passwd_hash
+            cracked_pw = self._crack_hash(passwd_entry, result_entry)
+            result_entry['cracked'] = bool(cracked_pw)
+        return result
+
+    def _crack_hash(self, passwd_entry, result_entry):
         with NamedTemporaryFile() as fp:
-            fp.write(b':'.join(passwd_entry[:2]))
-            fp.seek(0)
-            result_dict[key]['log'] = execute_shell_command('john --wordlist={} {}'.format(self.wordlist_path, fp.name))
-            output = execute_shell_command('john --show {}'.format(fp.name)).split('\n')
-        if len(output) > 2:
+            if result_entry['type'] is 'unix':
+                fp.write(b':'.join(passwd_entry[:2]))
+                fp.seek(0)
+                result_entry['log'] = execute_shell_command('{} --wordlist={} {}'.format(JOHN_PATH, self.wordlist_path, fp.name))
+                output = execute_shell_command('{} --show {}'.format(JOHN_PATH, fp.name)).split('\n')
+            elif result_entry['type'] is 'mosquitto':
+                fp.write(passwd_entry.encode())
+                fp.seek(0)
+                result_entry['log'] = execute_shell_command('{} {} --format=dynamic_82'.format(JOHN_PATH, fp.name))
```

Comment:
john should probably also use `--wordlist` in the mosquitto case

Reply:
what is more, the two cases can be combined by passing `b':'.join(passwd_entry[:2])` or `passwd_entry.encode()` to the function respectively and making the format a variable that is set according to the password type